charts/rstudio-connect/templates/_helpers.tpl

```diff
@@ -72,6 +72,9 @@ app.kubernetes.io/instance: {{ .Release.Name }}
     {{- else }}
       {{- $_ := set $launcherSettingsDict "KubernetesUseTemplates" "false" }}
     {{- end }}
+    {{- if .Values.launcher.templateValues.pod.serviceAccountName }}
+      {{- $_ := set $launcherSettingsDict "KubernetesDefaultServiceAccount" .Values.launcher.templateValues.pod.serviceAccountName }}
```

Comment:
This is potentially redundant with the above right? Basically we set the default in Connect _and_ in the `job.tpl`, so if Connect fails to send along the SA, then it would still get applied by the templates.

Reply:
Yes, there's definitely some awkwardness with the interaction between `Launcher.KubernetesDefaultServiceAccount` and `.Values.launcher.templateValues.pod.serviceAccountName`, and I'm not sure what the right answer is.

It's worth noting that when connect is deployed _outside_ of helm, if `Launcher.KubernetesDefaultServiceAccount` is unset, then Connect does not set a service account on the job that gets created, meaning Kubernetes should assign a default service account.